### PR TITLE
Remove usages of `failure` crate

### PR DESCRIPTION
**Pull request by jcape**
### Motivation

The `failure` crate has been [deprecated for over a year](https://github.com/rust-lang-nursery/failure/commit/135e2a3b9af422d9a9dc37ce7c69354c9b36e94b), this removes all our direct usages of it. Rust `nightly-2021-07-06` (wanted for M1) was throwing errors over this. Those warnings were later found to be [false positives](https://github.com/rust-lang/rust-clippy/issues/7434), but this work was already done, so we may as well move forward.

### In this PR
* Port all errors from `failure::Fail` to `displaydoc::Display`
* Fix invalid usage of `cargo trim` in CircleCI, intended to fix master CI

### CODE REVIEW

File: util/build/script/src/env.rs [hunk 1]
```diff
@@ -75,24 +73,17 @@ impl TryFrom<&str> for Endianness {
...
+    /// Endianness error: {0}
     Endianness(EndiannessError),
-    /// The target family could not be parsed
-    #[fail(display = "Target family error: {}", _0)]
+    /// Target family error: {0}
     TargetFamily(TargetFamilyError),
-    /// There was an error parsing a variable which was supposed to be a numeric
-    /// value
-    #[fail(display = "Could not parse {}: {}", _0, _1)]
+    /// Could not parse {0}: {1}"
```
Comment (eranrund) [inline util/build/script/src/env.rs:84]:
```suggestion
    /// Could not parse {0}: {1}
```

File: util/build/script/src/env.rs [hunk 2]
```diff
@@ -75,24 +73,17 @@ impl TryFrom<&str> for Endianness {
 }
 
 /// An enumeration of errors which can occur when parsing the build environment
-#[derive(Clone, Debug, Fail)]
+#[derive(Clone, Debug, Display)]
 pub enum EnvironmentError {
-    /// An environment variable which we expected was not set
-    #[fail(display = "Environment variable {} not readable: {}", _0, _1)]
+    /// Environment variable {0} not readable: {0}
```
Comment (eranrund) [inline util/build/script/src/env.rs:78]:
```suggestion
    /// Environment variable {0} not readable: {1}
```

File: ledger/sync/src/reqwest_transactions_fetcher.rs [hunk 3]
```diff
@@ -36,21 +36,21 @@ pub const DEFAULT_MERGED_BLOCKS_BUCKET_SIZES: &[u64] = &[10000, 1000, 100];
...
-    #[fail(display = "reqwest error on {}: {:?}", _0, _1)]
+    /// reqwest error on {0}: {1:?}
     ReqwestError(String, ReqwestError),
 
-    #[fail(display = "IO error on {}: {:?}", _0, _1)]
+    /// IO error on {0}: {1:?}
     IO(String, std::io::Error),
 
-    #[fail(display = "Received an invalid block from {}: {}", _0, _1)]
+    /// Received an invalid block from {0}: {1}"
```
Comment (eranrund) [inline ledger/sync/src/reqwest_transactions_fetcher.rs:50]:
```suggestion
    /// Received an invalid block from {0}: {1}
```

**Review by eranrund — APPROVED** [review]
Some small typos but other than that LGTM. Thank you!

**Review by christian-oudard — APPROVED** [review]

**Review by garbageslam — APPROVED** [review]

**Comment by garbageslam** [comment]
Thank you!

**Pull request merged**